### ISSUE TITLE
Can not snasphot layer

### ISSUE THREAD

**Issue by billypchan** [pointfreeco/swift-snapshot-testing#359]
Hi! When I rewrite one of my snapshot written with `FBSnapshotTestCase`, I find that the snapshot does not have the layer I added to an image view, but the old version of the snapshot (Wire-iOS Tests/ReferenceImages_64/Wire_iOS_Tests.FullscreenImageViewControllerSnapshotTests/testThatImageIsDarkenWhenSelectedByMenu@2x.png) has it:

I add the layer in this way:
```
let layer = CALayer()
layer.backgroundColor = UIColor.black.cgColor
imageView?.layer.insertSublayer(layer, at: 0)
```

And I snapshot the view controller:
```
verifySnapshot(matching: viewcontroller,
                                     as: .image)
```

The snapshot reference image do not have the black layer added.

Any suggestions for this case? Thanks in advance!

**Comment by darknoon** [comment]
Does the layer have a non-zero frame? Seems like there is some relevant code missing here.

**Comment by billypchan** [comment]
I check that the frame of the layer is set. 
see https://github.com/wireapp/wire-ios/blob/b331ea4b06652d01f9298b553d91e6818d41782a/Wire-iOS/Sources/UserInterface/Components/Controllers/FullscreenImageViewController.swift#L525

**Comment by billypchan** [comment]
And I also found that the blur layer can not be snapshotted, maybe also related to layer?

**Comment by darknoon** [comment]
@billypchan Yes, the way that snapshots are taken is not great… blur in general is not implemented in CoreGraphics drawing and because CA is composited in another process it is not easy to get a snapshot within the App's process that accurately represents the content. You can compare [this library](https://github.com/pointfreeco/swift-snapshot-testing/blob/main/Sources/SnapshotTesting/Snapshotting/CALayer.swift#L42) to [FBSnapshotTestCase](https://github.com/facebookarchive/ios-snapshot-test-case/blob/master/FBSnapshotTestCase/Categories/UIImage+Snapshot.m) and they are pretty similar.

When Apple frameworks like SceneKit need to render within the current process they use an API called [CARenderer](https://developer.apple.com/documentation/quartzcore/carenderer), which is public API on macOS but not on iOS (nonetheless, [it does exist](https://developer.limneos.net/?ios=13.1.3&framework=QuartzCore.framework&header=CARenderer.h)).

AFAICT APIs like [snapshotView(afterScreenUpdates:)](https://developer.apple.com/documentation/uikit/uiview/1622531-snapshotview) use a placeholder that CA can use to duplicate the content.
